lib/snyk/enrich_cyclonedx.go

```diff
@@ -60,7 +101,9 @@ func enrichCycloneDX(bom *cdx.BOM, logger zerolog.Logger) *cdx.BOM {
 					Msg("Could not identify package.")
 				return
 			}
-
+			for _, enrichFunc := range cdxEnrichers {
+				*component = enrichFunc(*component, purl)
```

Comment:
can we mutate rather than reassign?

Reply:
while we're at it should we also do the same for other cdx enrichers to keep things consistent? i.e `ecosystem` and `scorecard` cyclonedx enricher functions

Reply:
If we can, let's not touch it in this PR. Eventual consistency is great, but we can refactor in a later step.